mpt_api_client/resources/catalog/products.py

```diff
@@ -54,13 +57,38 @@ class ProductsServiceConfig:
 
 class ProductsService(
     PublishableMixin[Product],
-    ManagedResourceMixin[Product],
+    ModifiableResourceMixin[Product],
     CollectionMixin[Product],
     Service[Product],
     ProductsServiceConfig,
 ):
     """Products service."""
 
+    def create(
+        self,
+        resource_data: ResourceData,
+        icon: FileTypes,
+    ) -> Model:
+        """Create product with icon.
+
+        Args:
+            resource_data: Product data.
+            icon: Icon image in jpg, png, GIF, etc.
+
+        Returns:
+            Created resource.
+        """
+        files: dict[str, FileTypes] = {}
+        files["product"] = (
+            None,
+            json.dumps(resource_data),
+            "application/json",
+        )
+        files["icon"] = icon
```

Comment:
Why not doing in one step? it's easier to read and better  ->


```suggestion
        files: dict[str, FileTypes] = {
            "product": (None, json.dumps(resource_data), "application/json"),
            "icon": icon
        }
```